contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -2910,8 +2910,13 @@ pltsql_detect_numeric_overflow(int weight, int dscale, int first_block, int nume
 	if (sql_dialect != SQL_DIALECT_TSQL)
 		return false;
 
-	total_digit_count = (dscale == 0) ? (weight * numeric_base) :
-		((weight + 1) * numeric_base);
+	if (weight < 0)
```

Comment:
did any test fail due to this fix? also, we should add extensive tests for this around numeric overflow. Having both integer and scale being 1-4 5-8 9-12 and so on

Reply:
Nope.. We have added additional test cases. But you can check the table with `create table babel_4424_t6 (a numeric(10, 10));` where weight would be -1
<img width="228" alt="Screenshot 2023-11-10 at 10 19 36 AM" src="https://github.com/babelfish-for-postgresql/babelfish_extensions/assets/25386204/65838761-6713-4f41-b491-8d0412385983">
 and dscale = 10

Reply:
why total digit count is dscale + 1? it means we cannot store more than 37 fraction part?